src/tools/core/mode/mode_exec.cc

```diff
@@ -347,76 +486,276 @@ void ModeExecutive::setup_fcst_obs_data()
-      Fcst_sd.data.accum()                       != Obs_sd.data.accum()) {
+   // Store the grid
 
-      mlog << Warning << "\nsetup_fcst_obs_data() -> "
-           << "Forecast and observation accumulation times do not match "
-           << sec_to_hhmmss(Fcst_sd.data.valid()) << " != "
-           << sec_to_hhmmss(Obs_sd.data.valid()) << " for "
-           << engine.conf_info.Fcst->var_info->magic_str() << " versus "
-           << engine.conf_info.Obs->var_info->magic_str() << ".\n\n";
+   engine.set_grid(&grid);
+
+   // Regrid, if necessary
+
+   if ( !(obs_mtddf->grid() == grid) )  {
+      mlog << Debug(1)
+           << "Regridding observation " << engine.conf_info.Obs->var_info->magic_str()
+           << " to the verification grid.\n";
+      Obs_sd.data = met_regrid(Obs_sd.data, obs_mtddf->grid(), grid,
+                               engine.conf_info.Obs->var_info->regrid());
    }
 
+   // Rescale probabilites from [0, 100] to [0, 1]
+
+   if ( engine.conf_info.Obs->var_info->p_flag() ) rescale_probability(Obs_sd.data);
+
    mlog << Debug(1)
-        << "Forecast Field: "
-        << engine.conf_info.Fcst->var_info->name_attr() << " at "
-        << engine.conf_info.Fcst->var_info->level_attr()
-        << "\n"
         << "Observation Field: "
         << engine.conf_info.Obs->var_info->name_attr() << " at "
         << engine.conf_info.Obs->var_info->level_attr()
         << "\n";
 
-   // Mask out the missing data between fields
-
-   if(engine.conf_info.mask_missing_flag == FieldType_Fcst ||
-      engine.conf_info.mask_missing_flag == FieldType_Both)
-      mask_bad_data(Fcst_sd.data, Obs_sd.data);
-
-   // Mask out the missing data between fields
-
-   if(engine.conf_info.mask_missing_flag == FieldType_Obs ||
-      engine.conf_info.mask_missing_flag == FieldType_Both)
-      mask_bad_data(Obs_sd.data, Fcst_sd.data);
-
    // Parse the grid and/or polyline masks from the configuration
 
-   process_masks(Fcst_sd, Obs_sd);
+   process_obs_masks(Obs_sd);
 
    // Compute the min and max data values across both raw fields
 
-   Fcst_sd.data.data_range(fmin, fmax);
    Obs_sd.data.data_range(omin, omax);
-   if     (!is_bad_data(fmin) && !is_bad_data(omin)) data_min = min(fmin, omin);
-   else if(!is_bad_data(fmin) &&  is_bad_data(omin)) data_min = fmin;
-   else if( is_bad_data(fmin) && !is_bad_data(omin)) data_min = omin;
-
-   if     (!is_bad_data(fmax) && !is_bad_data(omax)) data_max = max(fmax, omax);
-   else if(!is_bad_data(fmax) &&  is_bad_data(omax)) data_max = fmax;
-   else if( is_bad_data(fmax) && !is_bad_data(omax)) data_max = omax;
+   data_min = omin;
+   data_max = omax;
 
    // Process percentile thresholds
 
-   engine.conf_info.set_perc_thresh(Fcst_sd.data, Obs_sd.data);
+   engine.conf_info.set_perc_thresh(Obs_sd.data);
 
    // store the input data units
-   funits = engine.conf_info.Fcst->var_info->units();
+   funits = "Not set";
```

Comment:
```suggestion
   funits = na_str;
```
Recommend never including whitespace in the units string because if its written to the `FCST_UNITS` output column it'll cause downstream parsing problems.